packages/devtools_app/lib/src/timeline/flutter/timeline_controller.dart

```diff
@@ -153,17 +220,15 @@ class TimelineController implements DisposableController {
     final uiThreadId = _threadIdForEvent(uiEventName, traceEvents);
     final gpuThreadId = _threadIdForEvent(gpuEventName, traceEvents);
 
-    _timelineModeNotifier.value = offlineData.timelineMode;
     offlineTimelineData = offlineData.shallowClone();
-    timeline
-      ..data = offlineData.shallowClone()
-      ..processor.primeThreadIds(
-        uiThreadId: uiThreadId,
-        gpuThreadId: gpuThreadId,
-      );
-    await timeline.processTraceEvents(traceEvents);
-
-    if (timeline.data.cpuProfileData != null) {
+    data = offlineData.shallowClone();
```

Comment:
why are there offlineTimelineData and data members with the same content?
consider removing offlineTimelineData and maybe just storing a bool that the data is offline?

Reply:
we need to store both so that we can revert to the original timeline data if event selections and frame selections while offline modify `data`.